src/afa/tests-afa.cc

```diff
@@ -11,11 +11,353 @@ using namespace Mata::Parser;
+    REQUIRE(std::to_string(aut.pre(Node{2}, 0).get_antichain()) == "{ { 2}}");
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{2}}, 0).get_antichain()) == "{ { 2}}");
+    REQUIRE(std::to_string(aut.pre(ClosedSet<State>(ClosedSet<State>::downward_closed, 0, 2, Nodes{Node{2}}), 0).get_antichain()) == "{ { 2}}");
+
+
+    REQUIRE(std::to_string(aut.pre(0, 1).get_antichain()) == "{ { 1}}");
+    REQUIRE(std::to_string(aut.pre(Node{0}, 1).get_antichain()) == "{ { 1}}");
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{0}}, 1).get_antichain()) == "{ { 1}}");
+    REQUIRE(std::to_string(aut.pre(ClosedSet<State>(ClosedSet<State>::downward_closed, 0, 2, Nodes{Node{0}}), 1).get_antichain()) == "{ { 1}}");
+
+    REQUIRE(std::to_string(aut.pre(1, 1).get_antichain()) == "{ { 0}}");
+    REQUIRE(std::to_string(aut.pre(Node{1}, 1).get_antichain()) == "{ { 0}}");
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{1}}, 1).get_antichain()) == "{ { 0}}");
+    REQUIRE(std::to_string(aut.pre(ClosedSet<State>(ClosedSet<State>::downward_closed, 0, 2, Nodes{Node{1}}), 1).get_antichain()) == "{ { 0}}");
+
+    REQUIRE(std::to_string(aut.pre(2, 1).get_antichain()) == "{ {}}");
+    REQUIRE(std::to_string(aut.pre(Node{2}, 1).get_antichain()) == "{ {}}");
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{2}}, 1).get_antichain()) == "{ {}}");
+    REQUIRE(std::to_string(aut.pre(ClosedSet<State>(ClosedSet<State>::downward_closed, 0, 2, Nodes{Node{2}}), 1).get_antichain()) == "{ {}}");
+
+
+    REQUIRE(std::to_string(aut.pre(Node{0, 1}, 0).get_antichain()) == "{ { 0, 2}}");
+    REQUIRE(std::to_string(aut.pre(Node{0, 2}, 0).get_antichain()) == "{ { 0, 2}}");
+    REQUIRE(std::to_string(aut.pre(Node{1, 2}, 0).get_antichain()) == "{ { 2}}");
+    REQUIRE(std::to_string(aut.pre(Node{0, 1, 2}, 0).get_antichain()) == "{ { 0, 2}}");
+
+    REQUIRE(std::to_string(aut.pre(Node{0, 1}, 1).get_antichain()) == "{ { 0, 1}}");
+    REQUIRE(std::to_string(aut.pre(Node{0, 2}, 1).get_antichain()) == "{ { 1}}");
+    REQUIRE(std::to_string(aut.pre(Node{1, 2}, 1).get_antichain()) == "{ { 0, 1}}");
+    REQUIRE(std::to_string(aut.pre(Node{0, 1, 2}, 1).get_antichain()) == "{ { 0, 1}}");
+
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{0}, Node{2}}, 0).get_antichain()) == "{ { 0}, { 2}}");
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{0}, Node{2}}, 1).get_antichain()) == "{ { 1}}");
+    REQUIRE(std::to_string(aut.pre(Nodes{Node{0}, Node{2}}).get_antichain()) == "{ { 0}, { 1}, { 2}}");
+
+} // }}}
+
+TEST_CASE("Mata::Afa antichain emptiness test")
+{
+    /////////////////////////////////
+    // Example of an automaton
+    /////////////////////////////////
+
+    Afa aut(3);
+
+	aut.initialstates = {0};
+	aut.finalstates = {2};
+
+    // TODO: Add transition and inverse transition simultaneously???
+    // Do we always care about inverse transitions (in context of other
+    // operations than backward emptiness test etc.?) Is it better
+    // to add inverse transitions on demand or always?
```

Comment:
We need to discuss this, in my opinion. Otherwise, this comment will be forgotten and left unresolved.

Reply:
Thanks for your suggestion, I have removed the comment and I have discussed it with others.